UnityProjectDP/Assets/Scripts/Visualization/UI/PopUps/AbstractTypePopUp.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using AnimArch.Visualization.Diagrams;
+using TMPro;
+using UnityEngine.UI;
+
+namespace AnimArch.Visualization.UI
+{
+    public abstract class AbstractTypePopUp : AbstractClassPopUp
+    {
+        private const string CUSTOM = "custom";
+        public TMP_Dropdown dropdown;
+        public TMP_Text customType;
+        public TMP_InputField customTypeField;
+        public Toggle isArray;
+        private readonly HashSet<TMP_Dropdown.OptionData> _variableData = new();
+
+        private void Awake()
+        {
+            dropdown.onValueChanged.AddListener(delegate
+            {
+                if (dropdown.options[dropdown.value].text == CUSTOM)
+                {
+                    customType.transform.gameObject.SetActive(true);
+                    customTypeField.transform.gameObject.SetActive(true);
+                }
+                else
+                {
+                    customType.transform.gameObject.SetActive(false);
+                    customTypeField.transform.gameObject.SetActive(false);
+                    customTypeField.text = "";
+                }
+            });
+        }
+
+        protected void SetType(string attributeType)
```

Comment:
Toto sa tyka aj return aj parameter type? Ak ano, tak by som to pomenaval mozno Set / Get ValueType. Ak iba navratoveho typu tak SetReturnType, ak iba attributu tak SetAttributeType.

Reply:
Týka sa to oboch, využíva sa to vždy keď sa setuje type (tj v AttributePopUp, MethodPopUp aj ParameterPopUp). Vždy ide o rovnaký princíp hence abstrakcia